packages/graphql/src/schema/resolvers/query/vector.ts

```diff
@@ -0,0 +1,102 @@
+ */
+
+import type { GraphQLFieldResolver, GraphQLResolveInfo, SelectionSetNode } from "graphql";
+import type { PageInfo } from "graphql-relay";
+import type { ConcreteEntityAdapter } from "../../../schema-model/entity/model-adapters/ConcreteEntityAdapter";
+import type { InterfaceEntityAdapter } from "../../../schema-model/entity/model-adapters/InterfaceEntityAdapter";
+import { translateRead } from "../../../translate";
+import type { VectorContext } from "../../../types";
+import { execute } from "../../../utils";
+import getNeo4jResolveTree from "../../../utils/get-neo4j-resolve-tree";
+import { isNeoInt } from "../../../utils/utils";
+import { createConnectionWithEdgeProperties } from "../../pagination";
+import type { Neo4jGraphQLComposedContext } from "../composition/wrap-query-and-mutation";
+
+export function vectorResolver({
+    vectorContext,
+    entityAdapter,
+}: {
+    vectorContext: VectorContext;
+    entityAdapter: ConcreteEntityAdapter | InterfaceEntityAdapter;
+}): GraphQLFieldResolver<any, any, any> {
+    return async function resolve(
+        _root: any,
+        args: any,
+        context: Neo4jGraphQLComposedContext,
+        info: GraphQLResolveInfo
+    ) {
+        context.vector = vectorContext;
+
+        const resolveTree = getNeo4jResolveTree(info, { args });
+        resolveTree.args.options = {
+            sort: resolveTree.args.sort,
+            limit: resolveTree.args.limit,
+            offset: resolveTree.args.offset,
+        };
+
+        // (context as Neo4jGraphQLTranslationContext).resolveTree = resolveTree;
+
+        const { cypher, params } = translateRead({
+            context: { ...context, resolveTree },
+            entityAdapter,
+            varName: "this",
+        });
+        const executeResult = await execute({
+            cypher,
+            params,
+            defaultAccessMode: "READ",
+            context,
+            info,
+        });
+
+        let totalCount = 0;
+        let edges: any[] = [];
+        let pageInfo: PageInfo = {
+            hasNextPage: false,
+            hasPreviousPage: false,
+            startCursor: null,
+            endCursor: null,
+        };
+
+        if (executeResult.records[0]) {
+            const record = executeResult.records[0].this;
+
+            totalCount = isNeoInt(record.totalCount) ? record.totalCount.toNumber() : record.totalCount;
+
+            const connection = createConnectionWithEdgeProperties({
+                selectionSet: resolveTree as unknown as SelectionSetNode,
+                source: { edges: record.edges },
+                args: { first: args.first, after: args.after },
+                totalCount,
+            });
+
+            edges = connection.edges as any[];
+            pageInfo = connection.pageInfo as PageInfo;
+        }
+
+        return {
+            [entityAdapter.operations.rootTypeFieldNames.connection]: {
+                totalCount,
+                edges,
+                pageInfo,
```

Comment:
Is it maybe better to have a constant somewhere which is like an "empty connection"? Then could just return that rather than defining with default values on like 69, and you could just return the connection as it is if there is a record in the result.